Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,431 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+
+/-- Constructor for `ReflexiveCofork` -/
+@[simps pt]
+def mk {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
+    ReflexiveCofork F where
+  pt := X
+  ι := reflexivePair.mkNatTrans π (F.map left ≫ π)
+
+@[simp]
+lemma mk_π {X : C} (π : F.obj zero ⟶ X) (h : F.map left ≫ π = F.map right ≫ π) :
+    (mk π h).π = π := rfl
+
+lemma condition (G : ReflexiveCofork F) : F.map left ≫ G.π = F.map right ≫ G.π := by
+  erw [Cocone.w G left, Cocone.w G right]
+
+@[simp]
+lemma app_one_eq_π (G : ReflexiveCofork F) : G.ι.app zero = G.π := rfl
+
+/-- The underlying `Cofork` of a `ReflexiveCofork`. -/
+abbrev toCofork (G : ReflexiveCofork F) : Cofork (F.map left) (F.map right) :=
+  Cofork.ofπ G.π (by simp)
+
+@[simp]
+lemma toCofork.π (G : ReflexiveCofork F) : Cofork.π (G.toCofork) = G.π := by
+  dsimp only [toCofork, Cofork.π]
+  aesop_cat
+
+end ReflexiveCofork
+
+noncomputable section
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+variable (F : WalkingReflexivePair ⥤ C)
+
+/-- Forgetting the reflexion yields an equivalence between cocones over a bundled reflexive pair and
+coforks on the underlying parallel pair. -/
+def inclusionWalkingReflexivePairEquivCocone :
+    Cocone F ≌ Cofork (F.map left) (F.map right) :=
+  (Functor.Final.coconesEquiv _ F).symm.trans
+    (Cocones.precomposeEquivalence
+      (diagramIsoParallelPair (WalkingParallelPair.inclusionWalkingReflexivePair ⋙ F))).symm
+
+@[inherit_doc inclusionWalkingReflexivePairEquivCocone]
+def inclusionWalkingReflexivePairEquiv_obj (G : ReflexiveCofork F) :
+    (inclusionWalkingReflexivePairEquivCocone F).functor.obj G ≅ G.toCofork :=
+  Cofork.ext (Iso.refl _)
+    (by simp [inclusionWalkingReflexivePairEquivCocone, Cofork.π])
+
+instance reflexivePair_hasColimit_of_hasCoequalizer
+    [h : HasCoequalizer (F.map left) (F.map right)] : HasColimit F := by
+  have := Limits.hasColimitOfIso (show _ ≅ parallelPair (F.map left) (F.map right) from
+    diagramIsoParallelPair (WalkingParallelPair.inclusionWalkingReflexivePair ⋙ F))
+  apply Functor.Final.hasColimit_of_comp WalkingParallelPair.inclusionWalkingReflexivePair
+
+lemma hasReflexiveCoequalizer_iff_hasCoequalizer :
+    HasColimit F ↔ HasCoequalizer (F.map left) (F.map right) := by
+  simpa only [hasColimit_iff_hasInitial_cocone]
+    using Equivalence.hasInitial_iff (inclusionWalkingReflexivePairEquivCocone F)
+
+/-- A reflexive cofork is a colimit cocone if and only if the underlying cofork is. -/
+def ReflexiveCofork.isColimitEquiv (G : ReflexiveCofork F) :
+    IsColimit (G.toCofork) ≃ IsColimit G :=
+  IsColimit.equivIsoColimit (inclusionWalkingReflexivePairEquiv_obj F G).symm|>.trans <|
+    (IsColimit.precomposeHomEquiv (diagramIsoParallelPair _).symm (G.whisker _)).trans <|
+      Functor.Final.isColimitWhiskerEquiv _ _
+
+/-- The colimit of a functor out of the walking reflexive pair is the same as the colimit of the
+underlying parallel pair. -/
+def inclusionWalkingReflexivePairColimitEquiv
```

Comment:
The name should contain `Iso`:
```suggestion
def reflexiveCoequalizerIsoCoequalizer
```

Reply:
You should also add two simp lemmas expressing compatibilities of this isomorphism with the canonical maps to the colimit and to the coequalizer. Names would be `ι_reflexiveCoequalizerIsoCoequalizer_hom` and `π_reflexiveCoequalizerIsoCoequalizer_inv`. (Same remarls for `ofIsReflexivePairColimitEquiv` below.)